### PR TITLE
Two more types of Legion and makes environment_smash levels into defines

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
One is another small legion, but it can infest corpses, meaning it will revive other legions if you don't kill it first.

The other is a legion that creates legions. I imagine it as something super edgy like a golem made up of legions melded together.

I need sprites for both of these

Added defines for environment_smash

### CODE REVIEW

**Comment by Repukan** [comment]
How about making tendrils spawn this in lavaland every so often?

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 1]
```diff
@@ -818,6 +819,57 @@
...
+	desc = "One of many."
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "legion"
+	icon_living = "legion"
+	icon_dead = "legion"
+	health = 450
+	maxHealth = 450
+	max_mobs = 3
+	spawn_time = 200
+	spawn_text = "peels itself off from"
```
Comment (XDTM) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: this feels so icky for some reason

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: A golem made of bodies is pretty icky yeah

**Comment by Mey-Ha-Zah** [comment]
I could have a crack at making a sprite... it's been a while, but I can give it a go.

**Comment by Militaires** [comment]
ill sprite the mega legion but ill make it 64x64 deal?

**Comment by KorPhaeron** [comment]
@Militaires sure

**Comment by KorPhaeron** [comment]
@Mey-Ha-Zah 

If you want to try making the once that can revive other legions sure. I was just going to use the current sprite but make it black or dark grey instead

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 2]
```diff
@@ -818,6 +819,57 @@
...
+	loot = list(/obj/item/organ/hivelord_core/legion = 3, /obj/effect/mob_spawn/human/corpse/damaged = 5)
+	move_to_delay = 14
+	vision_range = 5
+	aggro_vision_range = 9
+	idle_vision_range = 5
+	speed = 3
+	faction = list("mining")
+	weather_immunities = list("lava","ash")
+	obj_damage = 30
+	environment_smash = 1
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Is this a bool? If so make it true

Comment (GunHog) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: No, it has four (0-3) levels, with 3 being instant R-wall smashing.

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Someone really should make them defines

**Comment by Mey-Ha-Zah** [comment]
Alright, sprite preview: http://i.imgur.com/TYH0uIy.png

And all the sprites https://imgur.com/a/QyH37

**Comment by Militaires** [comment]
i memo'd you the real-resolution sprite on irc

this is meant to go with your LORE

**Comment by Militaires** [comment]
biglegion

**Comment by Repukan** [comment]
If I removed its head, would it die?

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Do the needful please Kor

**Comment by optimumtact** [comment]
Kor please

**Comment by Armhulen** [comment]
@Repukan
It would be very painful

**Comment by Repukan** [comment]
It's a big guy

**Comment by duncathan** [comment]
stale

**Comment by Militaires** [comment]
@duncathan @optimumtact the pr is fucking finished how is this stale

**Comment by Iamgoofball** [comment]
@Repukan for you

**Comment by Repukan** [comment]
Wait why did this get closed

**Comment by optimumtact** [comment]
It's not finished there is an unaddressed review comment.

**Comment by Iamgoofball** [comment]
it's a style thing, just do the needful with maintainer edit

**Comment by optimumtact** [comment]
no

**Comment by KorPhaeron** [comment]
@optimumtact 

Finished it

**Pull request merged**